library/x509_create.c

```diff
@@ -86,62 +92,161 @@ static const char *x509_at_oid_from_name( const char *name, size_t name_len )
     return( cur->oid );
 }
 
+static int x509_dn_check_special_chars( const char **cur, const char *end )
+{
+    const char *c = *cur;
+    int ret = 0;
+
+    /* Check for valid escaped characters */
+    if( c == end || *c != ',' )
+    {
+        ret = MBEDTLS_ERR_X509_INVALID_NAME;
+    }
+
+    return( ret );
+}
+
+static int dn_store_named_data( mbedtls_asn1_named_data **head, const char *oid,
+                                char data[MBEDTLS_X509_MAX_DN_NAME_SIZE], char *d,
+                                const char **s, const char **cur, const char *end )
+{
+    const char *c = *cur;
+
+    if( d - data == 0 )
+    {
+        return( MBEDTLS_ERR_X509_INVALID_NAME );
+    }
+
+    if( mbedtls_asn1_store_named_data( head, oid, strlen( oid ),
+                                       (unsigned char *) data,
+                                       d - data ) == NULL )
+    {
+        return( MBEDTLS_ERR_X509_ALLOC_FAILED );
+    }
+
+    /*
+     * Skip white spaces
+     */
+    while( c < end && *(c + 1) == ' ' )
+        c++;
+
+    *s = c + 1;
+
+    return ( 0 );
+}
+
 int mbedtls_x509_string_to_names( mbedtls_asn1_named_data **head, const char *name )
 {
     int ret = 0;
+    /*
+     * s points to the current string name.
+     * c points to the current character in the string being parsed.
```

Comment:
Can we rename `c` to `cur` to reflect this meaning?

Reply:
ok